internal/juju/jaas.go

```diff
@@ -142,3 +147,80 @@ func (jc *jaasClient) ReadRelations(ctx context.Context, tuple *JaasTuple) ([]Ja
 		}
 	}
 }
+
+func (jc *jaasClient) AddGroup(ctx context.Context, name string) (string, error) {
+	conn, err := jc.GetConnection(nil)
+	if err != nil {
+		return "", err
+	}
+	defer func() { _ = conn.Close() }()
+
+	client := jc.getJaasApiClient(conn)
+	req := params.AddGroupRequest{Name: name}
+
+	resp, err := client.AddGroup(&req)
+	if err != nil {
+		return "", err
+	}
+	return resp.UUID, nil
+}
+
+func (jc *jaasClient) ReadGroup(ctx context.Context, uuid string) (*JaasGroup, error) {
+	conn, err := jc.GetConnection(nil)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = conn.Close() }()
+
+	client := jc.getJaasApiClient(conn)
+	offset := 0
+	for {
+		req := &params.ListGroupsRequest{Offset: offset, Limit: 50}
+		groups, err := client.ListGroups(req)
```

Comment:
I think we need to rather add a new method to JIMM called ReadGroup() which accepts a UUID and returns the group info.

Reply:
True, would make this part easier to understand. The rest of the suggestions are already implemented.